org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/managers/GradlePreferenceChangeListener.java

```diff
@@ -42,23 +42,24 @@
 	public void preferencesChange(Preferences oldPreferences, Preferences newPreferences) {
 		ProjectsManager projectsManager = JavaLanguageServerPlugin.getProjectsManager();
 		if (projectsManager != null) {
-			if (hasAllowedChecksumsChanged(oldPreferences, newPreferences)) {
+			boolean gradleJavaHomeChanged = !Objects.equals(oldPreferences.getGradleJavaHome(), newPreferences.getGradleJavaHome());
+			if (hasAllowedChecksumsChanged(oldPreferences, newPreferences) || gradleJavaHomeChanged) {
```

Comment:
if (gradleJavaHomeChanged || hasAllowedChecksumsChanged(oldPreferences, newPreferences))

no need to compute checksum changes if we already know we need to bail

Reply:
Fixed.